cloud/ec2.go

```diff
@@ -652,102 +513,24 @@ func (m *ec2Manager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host, e
 		}))
 		return nil, errors.Wrap(err, msg)
 	}
-	if provider == onDemandProvider {
-		if err = m.spawnOnDemandHost(ctx, h, ec2Settings, blockDevices); err != nil {
-			msg := "error spawning on-demand host"
-			grip.Error(message.WrapError(err, message.Fields{
-				"message":       msg,
-				"host_id":       h.Id,
-				"host_provider": h.Distro.Provider,
-				"distro":        h.Distro.Id,
-			}))
-			return nil, errors.Wrap(err, msg)
-		}
-		grip.Debug(message.Fields{
-			"message":       "spawned on-demand host",
-			"host_id":       h.Id,
-			"host_provider": h.Distro.Provider,
-			"distro":        h.Distro.Id,
-		})
-	} else if provider == spotProvider {
-		err = m.spawnSpotHost(ctx, h, ec2Settings, blockDevices)
-		if err != nil {
-			msg := "error spawning spot host"
-			grip.Error(message.WrapError(err, message.Fields{
-				"message":       msg,
-				"host_id":       h.Id,
-				"host_provider": h.Distro.Provider,
-				"distro":        h.Distro.Id,
-			}))
-			if h.ShouldFallbackToOnDemand() && (strings.Contains(err.Error(), EC2InsufficientCapacity) || strings.Contains(err.Error(), EC2UnfulfillableCapacity)) {
```

Comment:
Also delete `ShouldFallbackToOnDemand` since it's unused.

Reply:
done